src/control/cmd/daos_server/storage.go

```diff
@@ -50,19 +49,38 @@ func (cmd *storageScanCmd) Execute(args []string) error {
 	}
 
 	cmd.log.Info("Scanning locally-attached storage...")
+
 	scanErrors := make([]error, 0, 2)
-	controllers, err := svc.ScanNvme()
-	if err != nil {
-		scanErrors = append(scanErrors, err)
-	} else {
-		cmd.log.Infof("NVMe SSD controller and constituent namespaces:\n%s", controllers)
-	}
+	e := make(chan error)
+	c := make(chan types.NvmeControllers)
+	m := make(chan types.ScmModules)
+
+	go func() {
+		controllers, err := svc.ScanNvme()
+		if err == nil {
+			c <- controllers
+		}
+		e <- err // last err value always processed after entries returned
+	}()
 
-	modules, err := svc.ScanScm()
-	if err != nil {
+	go func() {
+		modules, err := svc.ScanScm()
+		if err == nil {
+			m <- modules
+		}
+		e <- err
+	}()
+
+	select {
```

Comment:
I guess doing the scans concurrently could be more performant, but I'd recommend testing that assumption before making the implementation more complex.

In any case, I don't see how this would get all errors, since it only pulls from the channel once. Also, I think it's difficult to distinguish between the different scan errors since they're mixed together on one channel and the order they come back is nondeterministic. If you're going to make this concurrent, then I think you need a single channel given to both goroutines which accepts a results struct that looks something like this:

```go
type scanResult struct {
    err    error
    type   string
    result string
}

resCh := make(chan scanResult, 2)
```

In the goroutines, construct the result string to be displayed if there wasn't an error. After starting the goroutines, wait for the expected number of results, something like this:

```go
results := make([]scanResult, 0, 2)
for res := range results {
    results = append(results, res)
    if len(results) == cap(results) {
        close(resCh)
    }
}
```

Then you can pick through the results and display them as you like.